src/overlays/actors/ovl_En_Peehat/z_en_peehat.c

```diff
@@ -92,85 +99,987 @@ static ColliderJntSphInit D_80AD2790 = {
+        Matrix_RotateZ(-(this->jiggleRot * 0.1f), MTXMODE_APPLY);
+        Matrix_RotateY(-(this->jiggleRot * 0.13f), MTXMODE_APPLY);
+        Matrix_RotateX(-(this->jiggleRot * 0.115f), MTXMODE_APPLY);
+        gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_en_peehat.c", 1959),
+                  G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_OPA_DISP++, *dList);
+        Matrix_Pull();
+        CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_en_peehat.c", 1963);
+        return true;
+    }
+    return false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Peehat/func_80AD1BD8.s")
+void EnPeehat_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, void* thisx) {
+    static Vec3f D_80AD2844 = { 0, 0, 5500 };
+    static Vec3f D_80AD2850 = { 0, 0, -5500 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Peehat/EnPeehat_Update.s")
+    EnPeehat* this = THIS;
+    f32 damageYRot;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Peehat/func_80AD2224.s")
+    if (limbIndex == 4) {
+        Matrix_MultVec3f(&D_80AD2844, &this->bladeTip[0]);
+        Matrix_MultVec3f(&D_80AD2850, &this->bladeTip[1]);
+        return;
+    }
+    if (limbIndex == 3 && this->actor.params <= 0) {
+        damageYRot = 0.0f;
+        OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_peehat.c", 1981);
+        Matrix_Push();
+        Matrix_Translate(-1000.0f, 0.0f, 0.0f, MTXMODE_APPLY);
+        Collider_UpdateSpheres(0, &this->colJntSph);
+        Matrix_Translate(500.0f, 0.0f, 0.0f, MTXMODE_APPLY);
+        if (this->actor.colorFilterTimer != 0 && (this->actor.colorFilterParams & 0x4000)) {
+            damageYRot = Math_SinS(this->actor.colorFilterTimer * 0x4E20) * 0.35f;
+        }
+        Matrix_RotateY(3.2f + damageYRot, MTXMODE_APPLY);
+        Matrix_Scale(0.3f, 0.2f, 0.2f, MTXMODE_APPLY);
+        gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_en_peehat.c", 1990),
+                  G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_OPA_DISP++, *dList);
+        Matrix_Pull();
+        CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_en_peehat.c", 1994);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Peehat/func_80AD2438.s")
+void EnPeehat_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    static Vec3f D_80AD285C = { 0, 0, -4500 };
+    static Vec3f D_80AD2868 = { -4500, 0, 0 };
+    static Vec3f D_80AD2874 = { 4500, 0, 0 };
+    static Vec3f D_80AD2880 = { 0, 0, 4500 };
+    EnPeehat* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Peehat/EnPeehat_Draw.s")
+    func_80093D18(globalCtx->state.gfxCtx);
+    SkelAnime_DrawOpa(globalCtx, this->skelAnime.skeleton, this->skelAnime.jointTable, EnPeehat_OverrideLimbDraw,
+                      EnPeehat_PostLimbDraw, thisx);
```

Comment:
since skelanime switched to using void for the arg, we switched over to using `this` as what we pass to skelanime draw stuff